_source/_posts/2022-04-28-gdpr-and-app-security.md

```diff
@@ -0,0 +1,148 @@
+
+### Frontend cookie consent
+
+Users must be able to choose whether or not to accept cookies. GDPR emphasizes that users should be able to reject cookies easily. Some organizations make it [difficult to do this](https://www.zdnet.com/article/cookie-consent-most-websites-break-law-by-making-it-hard-to-reject-all-tracking/), but such underhanded tactics can result in heavy fines.
+
+A rule of thumb is to display the **Decline** button right next to the **Accept** button (and in the same font size) rather than offering a "Settings" or "Manage Cookies" link that makes it harder for users to refuse cookies and subtly bullies them into accepting the cookies on your platform. There are a ton of [resources](https://betterprogramming.pub/implement-a-cookie-consent-notification-within-5-minutes-82c845c55487) online that can help you implement GDPR-friendly frontend cookies. Compare your options so that you can adopt one that suits your organization best.
+
+### Data mapping
+
+[Data mapping](https://en.wikipedia.org/wiki/Data_mapping), or connecting information from multiple data sets, requires a top-down analysis of all of an organization's databases and what information they hold. This can be a big task, but it ensures transparency and accountability. Data mapping is required for the next steps.
+
+Data mapping can be manual, semi-automated, or automated. You should determine which approach works best for you, depending on the size of your company, your budget, the quality of your data management team, and the amount of data you work with. Deciding how to [implement data mapping](https://blog.hubspot.com/marketing/data-mapping) will require some understanding of your databases, as well as knowing which duplicated columns can be optimized to reduce redundancy.
+
+### Data purge functionality
+
+Previously, even after users requested that their data be deleted, organizations could remove it from public access and keep it somewhere hidden; however, the GDPR mandates that "officially" deleted data must be deleted everywhere. Organizations must ensure that they monitor data to delete it properly from all resources as needed.
+
+Data purge functionality [implementation](https://success.outsystems.com/Documentation/Best_Practices/Architecture/Data_Purging) will likely require some modification to your database schema so that you can identify data to mark for deletion, per user requests, and data to purge due to staleness. A `LastUpdatedDate` and `IsDeleted` column in your data table can be useful in identifying either type of data.
+
+### Refactoring DB schemas
+
+Data purge functionality may involve refactoring [database (DB) schemas](https://en.wikipedia.org/wiki/Database_schema) to add columns noting when a resource was created or whether it was deleted. These columns can be used to automate data purges based on whether the data has been held longer than needed, for instance.
```

Comment:
I'm not sure we need to link to Wikipedia here. Also, it seems like "Refactoring database schemas" might be a better heading.